File: SDKs/googleapis.com/androidpublisher/v3/java/lib/src/main/java/openapisdk/models/shared/RegionalSubscriptionOfferConfig.java
package openapisdk.models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

/**
 * RegionalSubscriptionOfferConfig
 * Configuration for a subscription offer in a single region.
**/
public class RegionalSubscriptionOfferConfig {
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("newSubscriberAvailability")
    public Boolean newSubscriberAvailability;
    public RegionalSubscriptionOfferConfig withNewSubscriberAvailability(Boolean newSubscriberAvailability) {
        this.newSubscriberAvailability = newSubscriberAvailability;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("regionCode")
    public String regionCode;
    public RegionalSubscriptionOfferConfig withRegionCode(String regionCode) {
        this.regionCode = regionCode;
        return this;
    }
}

File: SDKs/googleapis.com/androidpublisher/v3/python/src/sdk/models/shared/regionalsubscriptionofferconfig.py
import dataclasses
from typing import Optional
from dataclasses_json import dataclass_json
from sdk import utils


@dataclass_json
@dataclasses.dataclass
class RegionalSubscriptionOfferConfig:
    r"""RegionalSubscriptionOfferConfig
    Configuration for a subscription offer in a single region.
    """
    
    new_subscriber_availability: Optional[bool] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('newSubscriberAvailability') }})
    region_code: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('regionCode') }})
    
